### PR TITLE
Enhance the functions page

### PR DESCRIPTION
**Pull request by vindarel**
Time for basics !

Read it here: https://vindarel.github.io/cl-cookbook/functions.html VS https://lispcookbook.github.io/cl-cookbook/functions.html

### CODE REVIEW

File: data-structures.md [hunk 1]
```diff
@@ -1506,6 +1508,8 @@ With the `:include <struct>` argument:
 ;; #S(FEMALE :ID NIL :NAME "Lilie" :AGE NIL :GENDER "female")
 ~~~
 
+Note that the object system is more powerful.
```
Comment (pnathan) [inline data-structures.md]: the CLOS object system....

File: functions.md [hunk 2]
```diff
@@ -94,80 +385,115 @@ To simplify a bit, you can think of each symbol in CL having (at least) two "cel
...
+0
 ~~~
 
-Note that the argument to `funcall` (and `apply`) can either be the function itself, i.e. `#'+`, or a symbol which has the function in its function cell (is fbound to the function), i.e. `'+`.
 
+See more on [Practical Common Lisp](http://www.gigamonkeys.com/book/variables.html).
+
+## `setf` functions
 
-All of the above is _extremely simplified_ - we haven't even mentioned macros, special forms, symbol macros, self-evaluating objects, and lexical environments. Read the CLHS section about [form evaluation](http://www.lispworks.com/documentation/HyperSpec/Body/03_aba.htm) for the real deal.
```
Comment (pnathan) [inline functions.md]: I would move this into a Further Reading section, removing the "Extremely simplified" and doing some other wordsmithing.

**Comment by vindarel** [comment]
Another eye ?

**Comment by bendudson** [comment]
Looks good to me

**Comment by vindarel** [comment]
Let's go then, thanks.

**Pull request merged**